modules/basics/src/main/java/com/opengamma/strata/basics/index/FxIndex.java

```diff
@@ -47,7 +49,47 @@
   @FromString
   public static FxIndex of(String uniqueName) {
     ArgChecker.notNull(uniqueName, "uniqueName");
-    return extendedEnum().lookup(uniqueName);
+    try {
+      return extendedEnum().lookup(uniqueName);
+    } catch (IllegalArgumentException ex) {
+      CurrencyPair currencyPair = CurrencyPair.parse(uniqueName);
+      return FxIndex.of(currencyPair);
+    }
+  }
+
+  /**
+   * Obtains an instance from the specified currency pair.
+   * <p>
+   * If a currency pair does not have an implementation, an FX index will be created.
+   *
+   * @param currencyPair the currency pair
+   * @return the index
+   */
+  public static FxIndex of(CurrencyPair currencyPair) {
+    ArgChecker.notNull(currencyPair, "currencyPair");
+    return extendedEnum().lookupAll().values().stream()
+        .filter(index -> index.getCurrencyPair().equals(currencyPair))
+        .findFirst()
```

Comment:
Without a sort finding the first is essentially random. Maybe alpha sort by index name first (eg. there are two EUR/USD indices defined)